### ISSUE TITLE
Have you used 3d loss for fitting about simplify-X?

### ISSUE THREAD

**Issue by zhLawliet** [mks0601/I2L-MeshNet_RELEASE#84]
Thanks a lot for sharing the Simplify-X fits for H36M. Did you use 3d loss when fitting? I found that his side view is slanted, indicating that the depth is incorrect
![企业微信截图_da506ca2-92b9-4453-a86b-dff84b1d845e](https://user-images.githubusercontent.com/17424385/129006601-2ad08464-91c1-4de9-bfcd-35305fc044e0.png)

**Comment by zhLawliet** [comment]
thanks,i find you have  merged root pose and camera rotation.
        # merge root pose and camera rotation 
        root_pose = smpl_pose[self.root_joint_idx,:].numpy()
        root_pose, _ = cv2.Rodrigues(root_pose)
        root_pose, _ = cv2.Rodrigues(np.dot(R,root_pose))
        smpl_pose[self.root_joint_idx] = torch.from_numpy(root_pose).view(3)

 I try to understand what you mean is that the R  corresponds to the x-y view，the R is not suitable for z-y view?
if I want to show the z-y, I need other R?
the code is:

smpl_mesh_coord, smpl_joint_coord = self.smpl.layer['neutral'](smpl_pose, smpl_shape)
smpl_mesh_coord = smpl_mesh_coord.numpy().astype(np.float32).reshape(-1,3); 
fit_mesh_coord_cam = smpl_mesh_coord[...,[2,1,0]]  xyz->zyx
fit_mesh_coord_cam = (fit_mesh_coord_cam + 1)/2 * 255
vis(fit_mesh_coord_cam)

**Comment by mks0601** [comment]
I can't understand your question.. R is just a rotation matrix, included in the camera extrinsic parameter.

**Comment by mks0601** [comment]
How did you visualize your results?

**Comment by zhLawliet** [comment]
The code is for  side view:
        pose, shape, trans = smpl_param['pose'], smpl_param['shape'], smpl_param['trans']
        smpl_pose = torch.FloatTensor(pose).view(-1,3); smpl_shape = torch.FloatTensor(shape).view(1,-1); # smpl parameters (pose: 72 dimension, shape: 10 dimension)
        R, t = np.array(cam_param['R'], dtype=np.float32).reshape(3,3), np.array(cam_param['t'], dtype=np.float32).reshape(3) # camera rotation and translation
        # merge root pose and camera rotation 
        root_pose = smpl_pose[self.root_joint_idx,:].numpy()
        root_pose, _ = cv2.Rodrigues(root_pose)
        root_pose, _ = cv2.Rodrigues(np.dot(R,root_pose))
        smpl_pose[self.root_joint_idx] = torch.from_numpy(root_pose).view(3)
        smpl_mesh_coord, smpl_joint_coord = self.smpl.layer['neutral'](smpl_pose, smpl_shape)
        smpl_mesh_coord = smpl_mesh_coord.numpy().astype(np.float32).reshape(-1,3);
        fit_mesh_coord_cam = smpl_mesh_coord[...,[2,1,0]] #xyz->zyx
        fit_mesh_coord_cam = (fit_mesh_coord_cam + 1)/2 * 255
        fit_mesh_coord_cam = vis_mesh(img.copy(), fit_mesh_coord_cam, radius=1,color = (0,0,255),IS_cmap = False)

**Comment by mks0601** [comment]
Could you follow my codes in Human36M/Human36M.py?

**Comment by zhLawliet** [comment]
yes, I follow your codes in Human36M/Human36M.py,i can get right result about front view, which have applied extrinsic translation(R,T) and  internal parameters（cam_param['focal'], cam_param['princpt']）
![image](https://user-images.githubusercontent.com/17424385/129160568-977a7156-3ad4-4a16-9986-97a80e699d33.png)


<img width="556" alt="企业微信截图_593f1457-e879-441a-b59d-1a564cde4bd1" src="https://user-images.githubusercontent.com/17424385/129161036-c9d026d5-592a-43f2-981e-3c8d24ab393c.png">
the original coordinate system is x,y,z , this line is to convert the  coordinates  xyz->xzy for side view, I think the  smpl_mesh_coord have applied extrinsic translation by the "merge root pose and camera rotation",there is no  internal parameters for xzy, so I just want to visualize the orientation of the whole about side view.

**Comment by zhLawliet** [comment]
thanks,the 'internal parameters' is cam_param['focal'] and cam_param['princpt'],there  is  just  one extrinsics for front view, now I want to  visualize the orientation of the whole about side view. my unclear description may confuse you.
Maybe I need to change the question: How can I get the correct side view？

**Comment by mks0601** [comment]
The extrinsics are defined for all camera viewpoints.
You can apply extrinsics of the side viewpoint.

**Comment by zhLawliet** [comment]
Thanks for your  patient reply， I try it

**Comment by zhLawliet** [comment]
Thank you for your reply，your I2L-MeshNet wons the first and second place at 3DPW challenge on unknown assocation track which is not allowed to use ground truth data in any form, so how can you get right person  for multi person.  Another question is that  "bbox_root_pw3d_output.json" is just for 3DPW_test.json, but the above  3DPW challenge  use the entire dataset including its train, validation and test splits for evaluation.  so It's my pleasure that you can release this part of the code about the ECCV2020 3DPW challenge.

**Comment by mks0601** [comment]
Q. how can you get right person for multi person -> I used yolov5 human detector.
Q. I used param stage of I2L-MeshNet, so rootnet output is not required.

**Comment by zhLawliet** [comment]
thanks,I try it

**Comment by zhLawliet** [comment]
This should be a tracking issue，I want to reproduce your competition performance which wons the first and second place at 3DPW challenge on unknown assocation track. There are multiple candidate boxs in each frame by yolov4. How to choose the best matching box, especially for multiple people and scenes with overlapping characters. such as
![image_01265](https://user-images.githubusercontent.com/17424385/137420344-4895d1a5-7aba-480c-ad07-a9117d6ddbfc.jpg)
![image_00236](https://user-images.githubusercontent.com/17424385/137420371-cd5bac24-8dd3-40da-bb66-f12e624618a7.jpg)

**Comment by mks0601** [comment]
> Most of codes of the winning entry of 3DPW challenge is based on this repo. The tracking codes are newly added though.

we added human tracking codes as mentioned above.

**Comment by zhLawliet** [comment]
ok, thanks
